### PR TITLE
Add support for WIZNET5K ssl

### PR DESCRIPTION
**Pull request by justmobilize**
Updated to support SSL everywhere. To test you need:

cp: main ~~https://github.com/adafruit/circuitpython/pull/8954 (merged)~~
wiznet: main ~~https://github.com/adafruit/Adafruit_CircuitPython_Wiznet5k/pull/157 (merged)~~

If this looks good to people (and once the versions are determined), I would like to add tests before it's merged.

### CODE REVIEW

File: adafruit_connection_manager.py [hunk 1]
```diff
@@ -128,8 +128,28 @@ def get_radio_socketpool(radio):
...
 
-            # Note: SSL/TLS connections are not supported by the Wiznet5k library at this time
-            ssl_context = create_fake_ssl_context(pool, radio)
+            # Note: At this time, SSL/TLS connections are not supported by older
+            # versions of the Wiznet5k library or on boards withouut the ssl module
+            # see https://docs.circuitpython.org/en/latest/shared-bindings/support_matrix.html
+            ssl_context = None
+            try_ssl = False
+            cp_version = sys.implementation[1]
+            if pool.SOCK_STREAM == 1 and cp_version[0] >= 9:
```
Comment (justmobilize) [inline adafruit_connection_manager.py]: This was my best thought on checking the version. If there is a better way, let me know (I also hope ssl everywhere doesn't go out in a revision...)

**Comment by justmobilize** [comment]
@anecdata if you have time, would you be willing to try this?

**Comment by anecdata** [comment]
I'm having trouble getting the right pieces together for this. Here's what's installed:
`Adafruit CircuitPython 9.1.0-beta.0-27-g4ede914d45 on 2024-04-17; FeatherS3 with ESP32S3` (PR #8954)
`adafruit_requests` v3.2.5 (latest)
`adafruit_connection_manager` (this PR #9)
`adafruit_wiznet5k` v5.2.0 (includes #157, but not #151)

<details>
<summary>code...</summary>

```py
import traceback
import ssl
import board
import digitalio
from adafruit_wiznet5k.adafruit_wiznet5k import WIZNET5K
import adafruit_requests
import adafruit_connection_manager

TEXT_URL = "https://httpbin.org/get"

spi_bus = board.SPI()
cs = digitalio.DigitalInOut(board.A4)
mac = f'de:ad:be:ef:fe:{(252):02x}'  # I need this to avoid DHCP collisions
radio = WIZNET5K(spi_bus, cs, mac=mac, debug=False)
pool = adafruit_connection_manager.get_radio_socketpool(radio)
ssl_context = adafruit_connection_manager.get_radio_ssl_context(radio)
requests = adafruit_requests.Session(pool, ssl_context)

while True:
    print(f"Fetching from {TEXT_URL} ... ", end="")
    with requests.get(TEXT_URL) as resp:
        print(f'{resp.status_code}')
    time.sleep(5)
```
</details>

result:
```
Fetching from https://httpbin.org/get ... Traceback (most recent call last):
  File "code.py", line 23, in <module>
  File "adafruit_requests.py", line 591, in get
  File "adafruit_requests.py", line 525, in request
  File "adafruit_connection_manager.py", line 256, in get_socket
  File "adafruit_wiznet5k/adafruit_wiznet5k_socket.py", line 195, in getaddrinfo
  File "adafruit_wiznet5k/adafruit_wiznet5k_socket.py", line 210, in gethostbyname
AttributeError: 'NoneType' object has no attribute 'get_host_by_name'
```

I haven't used CM too much yet, maybe there's something wrong in the code.

**Comment by justmobilize** [comment]
@anecdata I found the issue (working on too many things at the same time). I'll update this PR tonight

**Comment by anecdata** [comment]
Yes, I fully understand, there are so many inter-related changes going on, it's hard to keep track of what goes with what.

**Comment by justmobilize** [comment]
Especially when it's one's own fault. I keep doing things here like I do professionally and thinking things will get merged and released quickly...

**Comment by justmobilize** [comment]
@anecdata this should work now... Sorry about that

**Comment by anecdata** [comment]
That did it... "200 OK" every 5 seconds :-)

**Review by anecdata — APPROVED** [review]

File: adafruit_connection_manager.py [hunk 2]
```diff
@@ -128,8 +130,22 @@ def get_radio_socketpool(radio):
...
+            ssl_context = None
+            cp_version = sys.implementation[1]
+            if pool.SOCK_STREAM == 1 and cp_version >= WIZNET5K_SSL_SUPPORT_VERSION:
+                try:
+                    import ssl  # pylint: disable=import-outside-toplevel
+
+                    ssl_context = ssl.create_default_context()
+                    pool.set_interface(radio)
+                except ImportError:
+                    """SSL not on board default to fake_ssl_context"""
```
Comment (dhalbert) [inline adafruit_connection_manager.py]:
```suggestion
                    # if SSL not on board, default to fake_ssl_context
```

**Review by dhalbert — CHANGES_REQUESTED** [review]
Not tested. Just one style thing.

**Comment by justmobilize** [comment]
@dhalbert updated

**Review by dhalbert — APPROVED** [review]
Thanks!

**Pull request merged**